File: Java/src/com/interpreter/box/BoxInstance.java
package com.interpreter.box;

import java.util.HashMap;            
import java.util.Map;                

class BoxInstance {                  
  private BoxClass klass;          
  private final Map<String, Object> fields = new HashMap<>();

  BoxInstance(BoxClass klass) {      
    this.klass = klass;              
  }                 
  
  Object get(Token name) {                           
	    if (fields.containsKey(name.lexeme)) {           
	      return fields.get(name.lexeme);                
	    }

	    BoxFunction method = klass.findMethod(this, name.lexeme);
	    if (method != null) return method;
	    
	    throw new RuntimeError(name, 
	        "Undefined property '" + name.lexeme + "'.");
	  }
  
  void set(Token name, Object value) {
	    fields.put(name.lexeme, value);   
	  }
  
  @Override                          
  public String toString() {         
    return klass.name + " instance"; 
  }         
}

File: C#/Interpreter/src/Box/BoxInstance.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Interpreter.Box
{
    public class BoxInstance
    {
        private BoxClass klass;
        private Dictionary<string, object> fields = new Dictionary<string, object>();

        public BoxInstance(BoxClass klass)
        {
            this.klass = klass;
        }

        public object Get(Token name)
        {
            if (fields.ContainsKey(name.lexeme))
            {
                return fields[name.lexeme];
            }

            BoxFunction method = klass.findMethod(this, name.lexeme);
            if (method != null) return method;

            Console.WriteLine("Undefined property '" + name.lexeme + "'.");

            return null;
        }

        void set(Token name, object value)
        {
            fields[name.lexeme] = value;
        }

        public string toString()
        {
            return klass.name + " instance";
        }
    }
}
